jcl/src/java.base/share/classes/java/lang/System.java

```diff
@@ -150,6 +150,11 @@
 		String definedFileEncoding = getSysPropBeforePropertiesInitialized(sysPropID_FileEncoding);
 		String definedOSEncoding = getSysPropBeforePropertiesInitialized(sysPropID_OSEncoding);
 		if (definedFileEncoding != null) {
+			/*[IF JAVA_SPEC_VERSION >= 18]*/
+			if ("COMPAT".equals(definedFileEncoding)) {
+				fileEncoding = platformEncoding;
```

Comment:
Is this going to be overwritten by later `fileEncoding = definedFileEncoding`?

Reply:
I'll remove this part and we can use your PR.

Reply:
Updated